### PR TITLE
Allow for a list and no searching for keys

### PR DESCRIPTION
**Pull request by scott-phillips-ah**
Searching for keys makes the secrets manager configuration require the ListSecrets permission, which some organizations don't like to grant to services. This PR allows users to define a list of secret ARNs to load, and it only loads those. Filtering still works, but I don't see a good reason to use a filter with a defined secrets list.

Test created and passed. Not sure the best way to develop an example

### CODE REVIEW

**Review by killianhale-earnin — APPROVED** [review]

**Comment by Kralizek** [comment]
Hi @scott-phillips-ah 

Thank you for your pull request.

I have two observations:
- can we avoid having a list of arns and a flag to set in the options? Basically instead of checking the flag we could check if the list is non empty. Otherwise the user could be misled into an erroneous path by either adding desired arns to the list or setting the flag but not both.
- i noticed you changed the styling of some files, please revert those changes.

Thanks.

File: src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs [hunk 1]
```diff
@@ -139,17 +139,23 @@ void SetData(IEnumerable<(string, string)> values, bool triggerReload)
...
         }
 
-        async Task<IReadOnlyList<SecretListEntry>> FetchAllSecretsAsync(CancellationToken cancellationToken)
+        private async Task<IReadOnlyList<SecretListEntry>> FetchAllSecretsAsync(CancellationToken cancellationToken)
         {
             var response = default(ListSecretsResponse);
 
             var result = new List<SecretListEntry>();
 
+            if (Options.UseDefinedArnList)
```
Comment (Kralizek) [inline src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs:148]: Check if the list contains items instead of using the flag

File: src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+using System.Collections.Generic;
 using Amazon.SecretsManager;
 using Amazon.SecretsManager.Model;
 
 namespace Kralizek.Extensions.Configuration.Internal
 {
     public class SecretsManagerConfigurationProviderOptions
     {
-        public Func<SecretListEntry, bool> SecretFilter { get; set; } = secret => true;
+        public bool UseDefinedArnList { get; set; } = false;
```
Comment (Kralizek) [inline src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs:10]: This one can be removed as we can infer its value from the amount of items in `DefinedSecretArns`.

File: src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
+        public List<string> DefinedSecretArns { get; set; } = new List<string>();
```
Comment (Kralizek) [inline src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs:11]: Keep spacing consistent.

File: tests/Tests.Extensions.Configuration.AWSSecretsManager/ConfigurationProviderExtensions.cs [hunk 4]
```diff
@@ -9,12 +9,7 @@ public static string Get(this IConfigurationProvider provider, params string[] p
         {
             var key = ConfigurationPath.Combine(pathSegments);
 
-            if (provider.TryGet(key, out var value))
-            {
-                return value;
-            }
-
-            return null;
+            return provider.TryGet(key, out var value) ? value : null;
```
Comment (Kralizek) [inline tests/Tests.Extensions.Configuration.AWSSecretsManager/ConfigurationProviderExtensions.cs:12]: Revert this change.

**Review by Kralizek — CHANGES_REQUESTED** [review]
As per the previous comment.
- Remove extra boolean flag
- Revert some code style changes.

File: src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs (hunk 3 above)
Comment (scott-phillips-ah) [reply src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs:10]: Done

File: tests/Tests.Extensions.Configuration.AWSSecretsManager/ConfigurationProviderExtensions.cs (hunk 4 above)
Comment (scott-phillips-ah) [reply tests/Tests.Extensions.Configuration.AWSSecretsManager/ConfigurationProviderExtensions.cs]: Done

File: src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs (hunk 2 above)
Comment (scott-phillips-ah) [reply src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs]: Good idea, done.

File: src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs (hunk 1 above)
Comment (scott-phillips-ah) [reply src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs]: `.Count > 0` Good idea, done.

**Comment by scott-phillips-ah** [comment]
Addressed all feedback, thanks for the quick review!

**Review by Kralizek — APPROVED** [review]

**Comment by Kralizek** [comment]
Hi @scott-phillips-ah, sorry I disappeared.

I'm not totally happy with the name of `DefinedSecretArns`. Can you suggest something else? I was more inclined towards `AcceptedSecretArns` but I welcome any other idea.

Also, would you be so kind to update the `README` file with this addition?

**Comment by scott-phillips-ah** [comment]
Hey, no worries. Good call on that. I changed it to what you specified, the alternative I'd use is `RequiredSecretArns` (which would indicate that the following secrets are **required** for the application to operate).

**Comment by Kralizek** [comment]
Would the application fail to start if a secret with one of the specified Arns were to be missing? If not, i don't think `Required` describes the behavior we are introducing.

**Comment by scott-phillips-ah** [comment]
In our case, it would. But I wouldn't say that is true in **every** case. We can leave it as it is.

**Comment by scott-phillips-ah** [comment]
Feel free to merge whenever.

**Comment by killianhale-earnin** [comment]
Any updates on this? I'd be happy to test or do whatever is needed to get this merged. I'm very excited for the feature ^_^

**Comment by Kralizek** [comment]
Hi @killianhale-earnin, could you take care of updating the readme? Also if you could try the feature in the wild, using one of the nightly builds, it would be great.

**Comment by scott-phillips-ah** [comment]
@killianhale-earnin @Kralizek  the README has been updated per your request.

**Review by Kralizek — APPROVED** [review]

**Comment by Kralizek** [comment]
@scott-phillips-ah @killianhale-earnin 

I'll release a new version once #47 is merged in :)
